src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala

```diff
@@ -560,18 +561,16 @@ final case class PurposeApiServiceImpl(
         tenant         <- tenantManagementService.getTenantById(organizationId)
         purposeUUID    <- purposeId.toFutureUUID
         purpose        <- purposeManagementService.getPurposeById(purposeUUID)
-        _              <- Future.successful(purpose).ensure(PurposeCannotBeCloned(purposeId))(isClonable)
-        dependencySeed = createPurposeSeed(purpose)
+        dailyCalls = getDailyCalls(purpose.versions)
+        _ <- Future.successful(purpose).ensure(PurposeCannotBeCloned(purposeId))(isClonable)
```

Comment:
don't use cats functions like `ensure`:

```suggestion
        _ <- if (isClonable) Future.successful(purpose) else Future.failed(PurposeCannotBeCloned(purposeId))
```